docs/architecture/accounts.md

```diff
@@ -1,211 +1,139 @@
-Accounts are basic building blocks representing a user or an autonomous smart contract.
+# Accounts
 
-For smart contracts, the go-to solution is account-based state. Miden supports expressive smart contracts via a Turing-complete language and the use of accounts.
+> The primary entities of the Miden protocol
 
-In Miden, an account is an entity that holds assets and defines rules about how to transfer these assets.
+## What is an Account?
 
-## Account design
+In Miden, an `Account` represents an entity capable of holding assets, storing data, and executing custom code. Each `Account` is essentially a specialized smart contract providing a programmable interface for interacting with its state and managed assets.
```

Comment:
Can you add here the "Why" as well? 

In Miden, we have accounts as entities because we want expressive smart contracts. 

For payments only we could do the UTXO model alone.

Reply:
I feel like the documentation should explain what and not really why. I don't think that it's relevant for the user / builder to understand our choice. It's more something that we would put in a blog post imo.

Reply:
Yes, I can see that argument. For me, at the beginning of the chapter, especially in a design / yellow paper section, we can have one paragraph of the why.